Equilobe.DailyReport.SL/AdvancedSettingsDataService.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Equilobe.DailyReport.DAL;
+using Equilobe.DailyReport.Models.Interfaces;
+using Equilobe.DailyReport.Models.Policy;
+using Equilobe.DailyReport.Models.Web;
+using Equilobe.DailyReport.Utils;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Equilobe.DailyReport.SL
+{
+    public class AdvancedSettingsDataService : IAdvancedSettingsDataService
+    {
+        #region IAdvancedSettingsDataService implementation
+        public List<SourceControlOptions> GetAllReposSourceControlOptions(long instanceId)
+        {
+            var advancedReportSettings = GetInstancePolicyStrings(instanceId);
+
+            if (!advancedReportSettings.Any())
+                return new List<SourceControlOptions>();
+
+            return advancedReportSettings
+                .Select(p => p.SourceControlOptions)
+                .ToList();
+        }
+
+        public List<User> GetUserMappings(long instanceId)
+        {
+            var advancedReportSettings = GetInstancePolicyStrings(instanceId);
+
+            if (!advancedReportSettings.Any())
+                return new List<User>();
+
+            return advancedReportSettings
+                .SelectMany(p => p.UserOptions)
+                .GroupBy(p => p.JiraUserKey)
```

Comment:
I think if you have the same user mapped on different repos with different usernames, you can lose them, if you just select the first case in the grouping. You should take only one user (so that we don't have duplicates) but also map it with all source control accounts that he has in the instance.